### PR TITLE
Moz/fix/values handling

### PR DESCRIPTION
**Pull request by Morriz**

### CODE REVIEW

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
This PR is not linked to the issue. 
It also contains various unrelated changes. Please split your commits to different PRs and explain the intent for each of them.

**Comment by Morriz** [comment]
done

File: .vscode/launch.json [hunk 1]
```diff
@@ -104,6 +104,7 @@
       "runtimeExecutable": "node",
       "runtimeArgs": ["--nolazy", "-r", "ts-node/register/transpile-only"],
       "args": ["src/otomi.ts", "--", "apply"],
+      // "args": ["src/otomi.ts", "--", "apply", "-l", "name=glasnostic"],
```
Comment (j-zimnowoda) [inline .vscode/launch.json:107]: This line could be removed

File: core.yaml [hunk 2]
```diff
@@ -16,6 +16,9 @@ k8s:
     - name: gatekeeper-system
       disableIstioInjection: true
     - name: gitea
+    - name: glasnostic
```
Comment (j-zimnowoda) [inline core.yaml:19]: The glanostic is also out of scope. Please move Glasnostic related changes to different PR

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]

**Review by j-zimnowoda — COMMENTED** [review]
Thanks for handling first time encryption issue.

File: src/common/values.ts [hunk 3]
```diff
@@ -55,12 +55,17 @@ let hasSops = false
...
 const writeValuesToFile = async (
   targetPath: string,
-  inValues: Record<string, any>,
+  inValues: Record<string, any> | undefined,
   overwrite = false,
 ): Promise<void> => {
-  const values = cloneDeep(inValues)
   const d = terminal('common:values:writeValuesToFile')
   const isSecretsFile = targetPath.includes('/secrets.') && hasSops
+  if (isEmpty(inValues) && isSecretsFile) {
```
Comment (j-zimnowoda) [inline src/common/values.ts:63]: I am wondering if it is better to perform this check after calling `removeBlankAttributes`

File: src/common/values.ts [hunk 4]
```diff
@@ -126,41 +131,43 @@ export const writeValues = async (values: Record<string, any>, overwrite = false
   // and write to their files
   const promises: Promise<void>[] = []
   if (settings) promises.push(writeValuesToFile(`${env.ENV_DIR}/env/settings.yaml`, settings, overwrite))
-  if (secretSettings)
+  if (secretSettings || overwrite)
```
Comment (j-zimnowoda) [inline src/common/values.ts:134]: Is it possible to move these checks inside writeValuesToFile?

Comment (Morriz) [reply src/common/values.ts:134]: no, as the input is dependent on it

File: src/common/values.ts (hunk 3 above)
Comment (Morriz) [reply src/common/values.ts:63]: good idea

**Review by j-zimnowoda — APPROVED** [review]
